clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3506,6 +3538,68 @@ protected Set<String> maybeSendIngestionHeartbeat() {
     return failedPartitions;
   }
 
+  /**
+   *  Resubscribe operation by passing new version role and partition role to {@link AggKafkaConsumerService}. The action
+   *  for leader and follower replica will be handled differently.
+   */
+  private void resubscribe(
+      PartitionConsumptionState partitionConsumptionState,
+      TopicPartitionReplicaRole.VersionRole versionRoleFromConsumerAction) throws InterruptedException {
+    if (isLeader(partitionConsumptionState)) {
+      resubscribeAsLeader(partitionConsumptionState, versionRoleFromConsumerAction);
+    } else {
+      resubscribeAsFollower(partitionConsumptionState, versionRoleFromConsumerAction);
+    }
+  }
+
+  protected void resubscribeAsFollower(
+      PartitionConsumptionState partitionConsumptionState,
+      TopicPartitionReplicaRole.VersionRole versionRoleFromConsumerAction) throws InterruptedException {
+    int partition = partitionConsumptionState.getPartition();
+    consumerUnSubscribe(versionTopic, partitionConsumptionState);
+    waitForAllMessageToBeProcessedFromTopicPartition(
+        new PubSubTopicPartitionImpl(versionTopic, partition),
+        partitionConsumptionState);
+    LOGGER.info(
```

Comment:
Shall we put the ingestion task name as prefix?

Reply:
I think SIT LOGGER will take ingestion task name by default like:
`2024-07-29 18:04:03 INFO [LeaderFollowerStoreIngestionTask] [venice-SIT-TestTopic_1c7b57bc6eb9a_4b836876_v1] Leader replica: TestTopic_1c7b57bc6eb9a_4b836876_v1-1 unsubscribe finished for future resubscribe.`